src/UCommerce.Transactions.Payments.Adyen/EventHandlers/AuthorisationEventHandler.cs

```diff
@@ -0,0 +1,36 @@
+using Adyen.Model.Notification;
+using Ucommerce.EntitiesV2;
+using Ucommerce.Pipelines;
+using Ucommerce.Transactions.Payments;
+
+namespace Ucommerce.Transactions.Payments.Adyen.EventHandlers;
+
+public class AuthorisationEventHandler : IEventHandler
+{
+    public bool CanHandle(string eventCode)
+    {
+        if (eventCode == "AUTHORISATION")
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    public void Handle(NotificationRequestItem notification, Payment payment)
+    {
+        payment.PaymentStatus = PaymentStatus.Get((int)PaymentStatusCode.Authorized);
```

Comment:
Instead of this servicelocator, I would like you to change it to ctor injection of the repo.